.github/workflows/haskell.yml

```diff
@@ -1,29 +1,29 @@
 name: Haskell CI
-on: [push]
+on: [push, pull_request]
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ghc: ["8.8.3"]
-        cabal: ["3.0"]
+        ghc: ["8.8.3", "9.2.1"]
```

Comment:
I would suggest to keep 8.10.7 and 9.2.1, and use only one Cabal

Reply:
build failed telling me it could not find 8.10.7, maybe the action uses an outdated version of `ghcup`? 🤔

Reply:
According to the docs, you should migrate to https://github.com/haskell/actions/tree/main/setup, the other one is no longer maintained